Neos.Media/Classes/Domain/Model/AssetCollection.php

```diff
@@ -177,4 +185,31 @@ public function removeTag(Tag $tag)
         }
         return false;
     }
+
+    public function getParent(): ?self
+    {
+        return $this->parent;
+    }
+
+    /**
+     * Throws an error if a circular dependency has been detected
+     * @throws \InvalidArgumentException
+     */
+    private function validateCircularHierarchy(): void
+    {
+        $parent = $this->parent;
+        $parents = [$parent];
```

Comment:
you never change that variable, can that work?

Reply:
I actually think this is correct as we have a circle and the parent would always appear twice, except if there was a circle to begin with and we would attach to that circle.